### ISSUE TITLE
I think integration is causing ha to crash 

### ISSUE THREAD

**Issue by milesroper** [faizpuru/ha-ambeo_soundbar#11]
Hi, since I enabled this integration ha has been locking up daily.  I’ve disabled it today and rebooted so will see if the issue reoccurs.  I can’t spot any issues in ha logs.  Any idea what info I can give you to have a look at this if when I’ve disabled it it stops crashing? 

I’m running raspberry pi4, latest ha versions.

**Comment by faizpuru** [comment]
Let's wait and see if it's related. Keep me informed. 
I never had the problem or any feedback.

If it's related I will investigate

**Comment by milesroper** [comment]
Hi, thanks for responding.  When I installed the integration after this ha was locking up twice a day.  It’s been 24 hours since I disabled it and haven’t had a crash.  So seems like it’s pointing in that direction.  Are you on the new release of hacs?  Just wondering if that might have something to do with it as that only came out a week or so ago.

**Comment by milesroper** [comment]
Just adding no issues still.  Any idea how when i turn it back on what info you might need to diagnose?  Thanks

**Comment by faizpuru** [comment]
I'm not available this weekend. I may have some time during the week.

**Comment by milesroper** [comment]
No worries.  Hang fire for a bit though.  I’ve think I’ve found I have a dodgy network cable.  I’ve replaced it tonight and we’ll see if no more issues then I’ll re enable integration and let you know.

**Comment by faizpuru** [comment]
The integration currently uses polling to retrieve data, resulting in many calls to the soundbar’s internal API. I will adjust the logic to use a coordinator, allowing the user to set the refresh interval.
I will try to do that during the week or over the weekend. (No guarantees)

**Comment by milesroper** [comment]
Hi, I’ve had the integration enabled for a couple of days and no issues.  I highly suspect the issue was the dodgy network cable.  As when I restarted ha I unplugged power supply and likely knocked cable which made it work again until it became loose and dropped out again.
 That’s why i couldn’t see any issues in logs, probably should have checked network interface stats.  Anyhow thought I better let you know and sorry for wasting your time.  So you close this issue or do I?  Thanks
